streaming/src/main/scala/org/apache/spark/streaming/dstream/FileInputDStream.scala

```diff
@@ -74,10 +74,12 @@ class FileInputDStream[K, V, F <: NewInputFormat[K,V]](
     directory: String,
     filter: Path => Boolean = FileInputDStream.defaultFilter,
     newFilesOnly: Boolean = true,
-    conf: Option[Configuration] = None)
+    @transient conf: Option[Configuration] = None)
```

Comment:
`Configuration` is used in `filesToRDD` function, so it is not just a constructor statements, if checkpoint is enable, this variable will be written into file and meet unserializable exception, you can check the linked JIRA for details. Sorry if I misunderstood anything.